core/templates/dev/head/pages/exploration_player/ConversationSkinDirective.js

```diff
@@ -228,551 +228,560 @@ oppia.animation('.conversation-skin-animate-cards', function() {
-        // side-by-side.
-        var TIME_PADDING_MSEC = 250;
-        var TIME_SCROLL_MSEC = 600;
-        var MIN_CARD_LOADING_DELAY_MSEC = 950;
-
-        var hasInteractedAtLeastOnce = false;
-        var _answerIsBeingProcessed = false;
-        var _nextFocusLabel = null;
-        // This variable is used only when viewport is narrow.
-        // Indicates whether the tutor card is displayed.
-        var tutorCardIsDisplayedIfNarrow = true;
-
-        $scope.explorationId = oppiaPlayerService.getExplorationId();
-        $scope.isInPreviewMode = oppiaPlayerService.isInPreviewMode();
-        $scope.isIframed = urlService.isIframed();
-        $rootScope.loadingMessage = 'Loading';
-        $scope.hasFullyLoaded = false;
-        $scope.recommendedExplorationSummaries = null;
-
-        $scope.OPPIA_AVATAR_IMAGE_URL = (
-          UrlInterpolationService.getStaticImageUrl(
-            '/avatar/oppia_avatar_100px.svg'));
-
-        $scope.activeCard = null;
-        $scope.numProgressDots = 0;
-
-        $scope.upcomingStateName = null;
-        $scope.upcomingContentHtml = null;
-        $scope.upcomingInlineInteractionHtml = null;
-
-        $scope.DEFAULT_TWITTER_SHARE_MESSAGE_PLAYER =
-          GLOBALS.DEFAULT_TWITTER_SHARE_MESSAGE_PLAYER;
-
-        $scope.getContentFocusLabel = function(index) {
-          return CONTENT_FOCUS_LABEL_PREFIX + index;
-        };
-
-        // If the exploration is iframed, send data to its parent about its
-        // height so that the parent can be resized as necessary.
-        $scope.lastRequestedHeight = 0;
-        $scope.lastRequestedScroll = false;
-        $scope.adjustPageHeight = function(scroll, callback) {
-          $timeout(function() {
-            var newHeight = document.body.scrollHeight;
-            if (Math.abs($scope.lastRequestedHeight - newHeight) > 50.5 ||
-                (scroll && !$scope.lastRequestedScroll)) {
-              // Sometimes setting iframe height to the exact content height
-              // still produces scrollbar, so adding 50 extra px.
-              newHeight += 50;
-              messengerService.sendMessage(messengerService.HEIGHT_CHANGE, {
-                height: newHeight,
-                scroll: scroll
-              });
-              $scope.lastRequestedHeight = newHeight;
-              $scope.lastRequestedScroll = scroll;
-            }
+oppia.directive('conversationSkin', [
+  '$scope', 'urlService', 'UrlInterpolationService',
+  function($scope, urlService, UrlInterpolationService) {
+    return {
+      restrict: 'E',
+      scope: {},
+      link: function(scope) {
+        var isIframed = urlService.isIframed();
+        scope.directiveTemplateUrl = isIframed ?
+          '/pages/exploration_player/conversation_skin_embed_directive.html' :
+          '/pages/exploration_player/conversation_skin_directive.html';
+      },
+      templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
```

Comment:
Here leave the previous version `template: '<div ng-include="directiveTemplateId"></div>',`